src/lib/core/partialcubeop.ts

```diff
@@ -0,0 +1,45 @@
+import { CubeIndex } from "lib/core/cubeindex";
+
+/**
+ * Computes the new (row, column) coordinates of the input CubeIndex after a
+ * clockwise rotation. Does not account for movements of indices on adjacent faces.
+ *
+ * {0, 0}, {0, 1}, {0, 2}      {2, 0}, {1, 0}, {0, 0}
+ * {1, 0}, {1, 1}, {1, 2}  ->  {2, 1}, {1, 1}, {0, 1}
+ * {2, 0}, {2, 1}, {2, 2}      {2, 2}, {1, 2}, {0, 2}
+ *
+ * @param cubeIndex The CubeIndex to be rotated
+ * @param dimension The dimension of the cube being acted on
+ */
+export function rotateClockwise(
+  cubeIndex: CubeIndex,
+  dimension: number,
+): CubeIndex {
+  return {
+    face: cubeIndex.face,
+    row: cubeIndex.column,
+    column: dimension - 1 - cubeIndex.row,
+  };
+}
+
+/**
+ * Computes the new (row, column) coordinates of the input CubeIndex after a
+ * counterclockwise rotation. Does not account for movements of indices on adjacent faces.
+ *
+ * {0, 0}, {0, 1}, {0, 2}      {0, 2}, {1, 2}, {2, 2}
+ * {1, 0}, {1, 1}, {1, 2}  ->  {0, 1}, {1, 1}, {2, 1}
+ * {2, 0}, {2, 1}, {2, 2}      {0, 0}, {1, 0}, {2, 0}
+ *
+ * @param cubeIndex The CubeIndex to be rotated
+ * @param dimension The dimension of the cube being acted on
+ */
+export function rotateCounterClockwise(
+  cubeIndex: CubeIndex,
```

Comment:
not really important. But if you want to be funky, you can destructure this like:
`{ face, column, row }: CubeIndex,`

Then refer directly to those properties below to reduce the code a tiny tiny bit

Reply:
Nice. Done.